### PR TITLE
notes for 18038

### PR DESCRIPTION
**Pull request by amitiuttarwar**

### CODE REVIEW

File: _posts/2020-04-29-#18038.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+host: amitiuttarwar
+status: upcoming
+commit:
+---
+
+## Notes
+- Today's PR has been extracted from
+  [#16698](https://github.com/bitcoin/bitcoin/pull/16698), which is a project
+  to improve privacy that involves moving rebroadcast functionality from the
+  wallet to the node. Having a smaller set of changes is always preferable
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:17]: maybe s/always/often/

File: _posts/2020-04-29-#18038.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  since it makes the PR easier to reason about and review. We discussed #16698
+  at a previous [review club](https://bitcoincore.reviews/16698.html).
+
+- On current master, the wallet has a method called `ResendWalletTransactions`
+  [link](https://github.com/bitcoin/bitcoin/blob/master/src/wallet/wallet.cpp#L1971).
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:22]: would suggest wrapping the link in parentheses

File: _posts/2020-04-29-#18038.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  This method is intended to _rebroadcast_ transactions, but also serves to
+  ensure a successful initial broadcast. Since only the wallet that originated
+  the txn will rebroadcast a transaction, there is a potential privacy leak
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:25]: maybe: s/txn will rebroadcast a transaction/the transaction will re-announce it/

File: _posts/2020-04-29-#18038.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  everytime `ResendWalletTransactions` runs.
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:26]: s/everytime/every time/

File: _posts/2020-04-29-#18038.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+- Today's PR reduces the frequency that `ResendWalletTransactions` runs from
+  ~30 min to every ~day. To do
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:29]: would write: "roughly every 30 minutes to approximately once a day"

File: _posts/2020-04-29-#18038.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  so, we extract the guarantees around initial broadcast.
+
+- Lets clarify some language:
+  - initial broadcast: when a node notifies the network about a transaction
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:33]: maybe s/notifies/first notifies/ (personally I like the word "announces" here, but ymmv)

File: _posts/2020-04-29-#18038.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  - rebroadcast: when a node sends subsequent notifications to the network
+    about that transaction
+  - unbroadcast (introduced in this pr): transactions where the initial
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:36]: s/pr/PR/ and s/transactions where/when/

File: _posts/2020-04-29-#18038.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  This method is intended to _rebroadcast_ transactions, but also serves to
+  ensure a successful initial broadcast. Since only the wallet that originated
+  the txn will rebroadcast a transaction, there is a potential privacy leak
+  everytime `ResendWalletTransactions` runs.
+
+- Today's PR reduces the frequency that `ResendWalletTransactions` runs from
+  ~30 min to every ~day. To do
+  so, we extract the guarantees around initial broadcast.
+
+- Lets clarify some language:
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:32]: maybe: "Let's define terms for 3 transaction broadcast states"

File: _posts/2020-04-29-#18038.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  - initial broadcast: when a node notifies the network about a transaction
+  - rebroadcast: when a node sends subsequent notifications to the network
+    about that transaction
+  - unbroadcast (introduced in this pr): transactions where the initial
+    broadcast hasn't been deemed successful
+
+- The mempool keeps track of transactions that it attempts to initially
+  broadcast. It uses a heuristic of receiving a relevant `GETDATA` to indicate
+  a successful initial broadcast. Every 10-15 minutes, it re-attempts broadcast
+  of the unbroadcast transactions (what a mouthful!)
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:42]: Could make clear whether this paragraph describes things before or after your PR.

File: _posts/2020-04-29-#18038.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+
+## Questions
+
+1. Did you review the PRs? [Concept ACK, approach ACK, tested ACK, or
```
Comment (jonatack) [inline _posts/2020-04-29-#18038.md:47]: PRs or PR?

**Comment by amitiuttarwar** [comment]
thanks for the review @jonatack. took most of your suggestions

**Comment by jonatack** [comment]
Thanks @amitiuttarwar, LGTM, merging and announcing the notes are up (and building your PR as I write this).

**Pull request merged**